engines/wage/combat.cpp

```diff
@@ -0,0 +1,914 @@
+bool WageEngine::handleAimCommand(const char *t) {
+	bool wasHandled = true;
+	Common::String target(t);
+
+	target.toLowercase();
+
+	if (target.contains("head")) {
+		_aim = Chr::HEAD;
+	} else if (target.contains("chest")) {
+		_aim = Chr::CHEST;
+	} else if (target.contains("side")) {
+		_aim = Chr::SIDE;
+	} else {
+		wasHandled = false;
+		appendText("Please aim for the head, chest, or side.");
+	}
+
+	_commandWasQuick = true;
+
+	return wasHandled;
+}
+
+bool WageEngine::handleWearCommand(const char *t) {
+	Chr *player = _world->_player;
+	char buf[512];
+	Common::String target(t);
+	bool handled = false;
+
+	target.toLowercase();
+
+	for (ObjArray::const_iterator it = _world->_player->_inventory.begin(); it != _world->_player->_inventory.end(); ++it) {
+		Common::String n((*it)->_name);
+
+		if (target.contains(n)) {
+			if ((*it)->_type == Obj::HELMET) {
+				wearObj(*it, Chr::HEAD_ARMOR);
+			} else if ((*it)->_type == Obj::CHEST_ARMOR) {
+				wearObj(*it, Chr::BODY_ARMOR);
+			} else if ((*it)->_type == Obj::SHIELD) {
+				wearObj(*it, Chr::SHIELD_ARMOR);
+			} else if ((*it)->_type == Obj::SPIRITUAL_ARMOR) {
+				wearObj(*it, Chr::MAGIC_ARMOR);
+			} else {
+				appendText("You cannot wear that object.");
+			}
+
+			handled = true;
+			break;
+		}
+	}
+
+	for (ObjList::const_iterator it = player->_currentScene->_objs.begin(); it != player->_currentScene->_objs.end(); ++it) {
+		Common::String n((*it)->_name);
+		n.toLowercase();
+		if (target.contains(n)) {
+			snprintf(buf, 512, "First you must get the %s.", (*it)->_name.c_str());
+			appendText(buf);
+
+			handled = true;
+			break;
+		}
+	}
+
+	return handled;
+}
+
+void WageEngine::wearObj(Obj *o, int pos) {
+	Chr *player = _world->_player;
+	char buf[512];
+
+	if (player->_armor[pos] == o) {
+		snprintf(buf, 512, "You are already wearing the %s.", o->_name.c_str());
+		appendText(buf);
+	} else {
+		if (player->_armor[pos] != NULL) {
+			snprintf(buf, 512, "You are no longer wearing the %s.", player->_armor[pos]->_name.c_str());
+			appendText(buf);
+		}
+
+		player->_armor[pos] = o;
+		snprintf(buf, 512, "You are now wearing the %s.", o->_name.c_str());
+		appendText(buf);
+	}
+}
+
+
+bool WageEngine::handleOfferCommand(const char *target) {
+	Chr *player = _world->_player;
+	Chr *enemy = getMonster();
+
+	if (enemy != NULL) {
+		Common::String t(target);
+		t.toLowercase();
+
+		for (ObjArray::const_iterator it = player->_inventory.begin(); it != player->_inventory.end(); ++it) {
+			Common::String n((*it)->_name);
+			n.toLowercase();
+
+			if (t.contains(n)) {
+				if ((*it)->_value < enemy->_rejectsOffers) {
+					appendText("Your offer is rejected.");
+				} else {
+					appendText("Your offer is accepted.");
+					appendText(enemy->_acceptsOfferComment.c_str());
+					_world->move(*it, enemy);
+					_world->move(enemy, _world->_storageScene);
+				}
+
+				return true;
+			}
+		}
+	}
+
+	return false;
+}
+
+bool WageEngine::tryAttack(Obj *weapon, Common::String &input) {
```

Comment:
(Minor) This can also take `const` versions of both parameters. Might even make sense to make `weapon` a reference, but as far as I understand it, the only caller has a pointer anyway, so this isn't too important. But in general using (const) references can clarify that the parameter should always be non-`NULL`.